### PR TITLE
Invalid Blik code redirects to error page

### PR DESCRIPTION
**Pull request by coldic3**
image

### CODE REVIEW

File: src/Controller/DisplayPaymentFailedPageAction.php [hunk 1]
```diff
@@ -22,12 +23,24 @@ public function __construct(
     public function __invoke(Request $request, string $orderToken): Response
     {
         $order = $this->findOrderOr404($orderToken);
-        $payment = $order->getLastPayment();
+        $newPayment = $order->getLastPayment();
+        $failedPayment = $order->getPayments()->get($order->getPayments()->count() - 2);
```
Comment (lchrusciel) [inline src/Controller/DisplayPaymentFailedPageAction.php:27]: What kind of sorcery is it?

Comment (coldic3) [reply src/Controller/DisplayPaymentFailedPageAction.php:27]: I forgot to refactor this, but the main idea should stay I believe. We need the error message from the failed payment. Right now, the last payment is a new payment, not the one that failed. That’s why I take one more payment behind.

Comment (lchrusciel) [reply src/Controller/DisplayPaymentFailedPageAction.php:27]: we can tak last failed payment, as there is option to pass argument to that function

Comment (coldic3) [reply src/Controller/DisplayPaymentFailedPageAction.php]: fixed

**Comment by lchrusciel** [comment]
wywalił mi angielski błąd, ale chyba działa
image

**Review by lchrusciel — APPROVED** [review]

**Pull request merged**